controller/controller.go

```diff
@@ -200,12 +214,15 @@ func (c *Impl) processNextWorkItem() bool {
 		// Run Reconcile, passing it the namespace/name string of the
 		// resource to be synced.
 		if err := c.Reconciler.Reconcile(ctx, key); err != nil {
+			c.statsReporter.ReportReconcile(time.Now().Sub(startTime), key, falseString)
 			return fmt.Errorf("error syncing %q: %v", key, err)
 		}
+
 		// Finally, if no error occurs we Forget this item so it does not
 		// get queued again until another change happens.
 		c.WorkQueue.Forget(obj)
 		c.logger.Infof("Successfully synced %q", key)
+		c.statsReporter.ReportReconcile(time.Now().Sub(startTime), key, trueString)
```

Comment:
I wonder if we should just use:
```go
var err error
defer func() {
   status := "true"
   if err != nil {
     status = "false"
   }
   c.statsReporter.ReportReconcile(time.Now().Sub(startTime), key, status)
}()
... // paths that may set/return err
```

So that we cannot introduce new paths that don't record metrics.